lms/djangoapps/commerce/api/v1/views.py

```diff
@@ -64,10 +73,17 @@ class OrderView(APIView):
     """ Retrieve order details. """
 
     authentication_classes = (SessionAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticatedOrActivationOverridden,)
```

Comment:
If we can achieve our desired outcome without this change then I'm all for removing it!